providers/legacyprov.c

```diff
@@ -184,11 +184,34 @@ int OSSL_provider_init(const OSSL_PROVIDER *provider,
 
     *out = legacy_dispatch_table;
 
+#if 1
     /*
      * We want to make sure that all calls from this provider that requires
      * a library context use the same context as the one used to call our
      * functions.  We do that by passing it along as the provider context.
      */
-    *provctx = c_get_libctx(provider);
+
+    /*
+     * TODO(3.0): FIXME FIXME FIXME DANGER DANGER DANGER
+     * 
+     * We should really replace this with the code below - but for some reason
+     * I'm getting a linker error when I do so.
```

Comment:
Interesting...  I'm about to look at the same thing, so I can investigate as well.